cloudbot/util/decorators.py

```diff
@@ -0,0 +1,25 @@
+from time import time
+
+SPAM_LIST = {}
+
+
+def spam_guard(func):
+    def wrapper(*args, **kwargs):
+        nick = kwargs.get('nick', None)  # pull the nick from the args
+        if nick:  # if there is no nick specified for whatever reason, carry on
+            global SPAM_LIST
+            if SPAM_LIST.get(nick, None):
+                SPAM_LIST[nick][0] = time()
+                if time() - SPAM_LIST[nick][0] < SPAM_LIST[nick][1]:
```

Comment:
Maybe:

```python
current_time = time()
if (current_time - SPAM_LIST[nick][0]) < SPAM_LIST[nick][1]:
    # ... code ...
else:
    # ... attempted to execute a command ...

SPAM_LIST[nick][0] = current_time
```
?

As it is, setting it to time() before checking the delay neutralizes the function...